FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        assert( pPublishInfo != NULL );
+
+        for( i = 0; i < SUBSCRIPTIONS_MAX_COUNT; i++ )
+        {
+            if( pxSubscriptions[ i ].topicFilterLength > 0 )
+            {
+                xIsMatched = topicFilterMatch( pPublishInfo->pTopicName,
+                                               pPublishInfo->topicNameLength,
+                                               pxSubscriptions[ i ].pTopicFilter,
+                                               pxSubscriptions[ i ].topicFilterLength );
+
+                if( xIsMatched )
+                {
+                    LogInfo( ( "Adding publish to response queue for %.*s",
+                               pxSubscriptions[ i ].topicFilterLength,
+                               pxSubscriptions[ i ].pTopicFilter ) );
+                    prvCopyPublishToQueue( pPublishInfo, pxSubscriptions[ i ].pResponseQueue );
+                }
+            }
+        }
+    }
+    else
+    {
+        /* Handle other packets. */
+        switch( pPacketInfo->type )
+        {
+            case MQTT_PACKET_TYPE_PUBACK:
+            case MQTT_PACKET_TYPE_PUBCOMP:
+                xAckInfo = prvPopAck( packetIdentifier );
+
+                if( xAckInfo.packetId == packetIdentifier )
+                {
+                    xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                    if( xAckInfo.callback != NULL )
+                    {
+                        xAckInfo.callback( xAckInfo.pCommandContext );
+                    }
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_SUBACK:
+                xAckInfo = prvPopAck( packetIdentifier );
+
+                if( xAckInfo.packetId == packetIdentifier )
+                {
+                    pxSubscribeInfo = xAckInfo.pCommandContext->pSubscribeInfo;
+
+                    for( i = 0; i < xAckInfo.pCommandContext->subscriptionCount; i++ )
+                    {
+                        LogInfo( ( "Adding subscription to %.*s",
+                                   pxSubscribeInfo[ i ].topicFilterLength,
+                                   pxSubscribeInfo[ i ].pTopicFilter ) );
+                        LogInfo( ( "Filter length: %u", pxSubscribeInfo[ i ].topicFilterLength ) );
+                        prvAddSubscription( pxSubscribeInfo[ i ].pTopicFilter,
+                                            pxSubscribeInfo[ i ].topicFilterLength,
+                                            xAckInfo.pCommandContext->pResponseQueue );
+                    }
+                }
+                else
+                {
+                    xStatus = MQTTBadResponse;
+                }
+
+                xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                if( xAckInfo.callback != NULL )
```

Comment:
How about if xStatus = MQTTBadResponse?

Reply:
I don't think xStatus can be MQTTBadResponse